client/my-sites/domains/domain-management/list/domain-item.jsx

```diff
@@ -3,31 +3,124 @@
  */
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import Gridicon from 'components/gridicon';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import { CompactCard } from '@automattic/components';
+import { Button, CompactCard } from '@automattic/components';
+import FormCheckbox from 'components/forms/form-checkbox';
+import DomainNotice from 'my-sites/domains/domain-management/components/domain-notice';
+import EllipsisMenu from 'components/ellipsis-menu';
+import PopoverMenuItem from 'components/popover/menu-item';
 
 class DomainItem extends PureComponent {
 	static propTypes = {
 		domain: PropTypes.object.isRequired,
+		isManagingAllSites: PropTypes.bool,
+		showSite: PropTypes.bool,
+		showCheckbox: PropTypes.bool,
 		onClick: PropTypes.func.isRequired,
+		onAddEmailClick: PropTypes.func.isRequired,
+		onToggle: PropTypes.func,
+	};
+
+	static defaultProps = {
+		isManagingAllSites: false,
+		showSite: false,
+		showCheckbox: false,
+		onToggle: null,
 	};
 
 	handleClick = () => {
 		this.props.onClick( this.props.domain );
 	};
 
+	stopPropagation = ( event ) => {
+		event.stopPropagation();
+	};
+
+	onToggle = ( event ) => {
+		if ( this.props.onToggle ) {
+			this.props.onToggle( event.target.checked );
+		}
+	};
+
+	addEmailClick = ( event ) => {
+		const { domain, onAddEmailClick } = this.props;
+		event.stopPropagation();
+		onAddEmailClick( domain );
+	};
+
+	renderOptionsButton() {
+		const { isManagingAllSites, translate } = this.props;
+
+		return (
+			<div className="domain-item__options">
+				<EllipsisMenu onClick={ this.stopPropagation } toggleTitle={ translate( 'Options' ) }>
+					{ ! isManagingAllSites && (
+						<PopoverMenuItem icon="domains">{ translate( 'Make primary domain' ) }</PopoverMenuItem>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **46** times:
`translate( 'Primary Domain' )` *ES Score: 6*